database/buildSchema/22_action_plugin.sql

```diff
@@ -1,12 +1,14 @@
 -- action plugins
 
 CREATE TABLE public.action_plugin (
---    id serial primary key,
     code varchar primary key,
     name varchar,
+    type varchar,
```

Comment:
This is coming from the `plugin.json` config file. Since the INSERT now accepts all fields that are part of the plugin this was one of them. I also though it was odd to have that there. I think we have 2 options:
1. Remove this type form the plugin.json file.
2. Remove the type form the plugin before sending it to the DB.

Reply:
I don't think we want to put all fields from the `plugin.json` file into the database. Apart from the compulsory ones, a plugin author should be able to add any fields they want, for other purposes (such as some other software that might want to catalogue them, I don't know, but they should be pretty open-ended), and we don't want to cause a database error whenever a field is added that isn't in the database already. The database fields were only intended to capture the essential information that the system needs to run the plugin, whereas the `plugin.json` is open to being a much more comprehensive metadata store (and there definitely isn't a one-to-one mapping between meta-data and database fields (e.g. the `path` field is a combination of the discovered path (not in `plugin.json`) and the filename.))

Reply:
Yes, I agree. I'll remove this fields from the DB now and documentation.

I think the ideal would be that only the fields expected to be included in the database will be passed to the method. I'll check how this can be done using Typescript and casting. I'm sure there is a simple way to do this.